src/main/java/UtilsService.java

```diff
@@ -0,0 +1,120 @@
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class UtilsService {
+
+    public boolean isValidJson(String responseText) {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.readTree(responseText);
+            return true;
+        } catch (Exception e) {
+            log.error("Invalid JSON response: " + responseText);
```

Comment:
Logging the entire response text could expose sensitive information. Consider logging a sanitized version or a message indicating the error without revealing the input. (Severity: High)